proposals/0052-experimental-dxil-ops.md

```diff
@@ -221,16 +357,59 @@ partition for their own use without collision with other opcodes.
 Pros:
  * Fairly simple
  * Quick to implement
+ * Could be implemented "by hand" today by hard coding opcodes in clang,
+   DXC requires some updates to opcode generation code.
  * Enables basic opcodes extension system
 Cons:
- * transistion from experimental to stable isn't just clearing the partition
-   * other stable ops may have already taken that number
-   * significantly complicates the experimental->stable transition
+ * transistion from experimental-> stable requires manual renumbering which
+   will change the lowering.
+
+### Top 16 bits as opcode partition
+Identical concept as the 1 bit proposal with a couple key changes.
+
+```
+0b 0000 0000 0000 0000 0000 0000 0000 0000
+                       ^^^^ ^^^^ ^^^^ ^^^^ opcode space
+   ^^^^ ^^^^ ^^^^ ^^^^-------------------- opcode partition
+```
+
+This will create ~64k partitions each with ~64k opcodes.
 
-### Top 16 bits as "opcode partition" value
-Identical concept as above but with 64k partitions, each with 64k opcodes.
+The opcode partition is now large enough that it earns a special name
+`FeatureID`. Along with this name change, the intended usage also changes.
+Each new feature in development reserves a new FeatureID as the first step
+in the development lifecycle. This enables async work as coordination is only
+required at the FeatureID scope which can be atomically reserved. The one
+exception is FeatureID `0x0000` which is reserved for past stable opcodes to 
+maintain their current value. 
+
+The validator or some other mechanism must maintain a list of experimental
+FeatureIDs. When an experimental FeatureID is used the entire shader must
+be marked as preview.
+
+FeatureIDs may be reused once a FeatureID has been marked reserved for at least
+one shader model. Ex: Feature ID `0xDEADBEEF` is introduced in 6.8 as
+experimental, marked as reserved for 6.9, so it may be recycled in 6.10
+
+Pros:
+ * Could be implemented "by hand" today by hard coding opcodes in clang,
+   DXC requires some updates to opcode generation code.
+ * Enables basic opcodes extension system
+ * Very flexible for opcode asignments
+ * Decreases the occurance of opcode collison merge conflicts
```

Comment:
This also allows for external parties to privately experiment with new DXIL operations before proposing features, without having to constantly update those opcodes with breaking changes every time a new experimental op is added to main. Then transitioning to a public proposal requires far fewer changes than transitioning from other hacky methods of extending DXIL for such private experiments today.

However, another solution for that using a single experimental table would be to reserve a bunch of opcodes for such purposes and just never reuse those. Still, we would need to anticipate and reserve sufficient opcodes for this use case.